src/ui/Omnibox/OmniboxAnalytics.ts

```diff
@@ -0,0 +1,70 @@
+import { compact, filter, map, reduceRight, last } from 'underscore';
+import { IAnalyticsOmniboxSuggestionMeta } from '../Analytics/AnalyticsActionListMeta';
+
+export class OmniboxAnalytics {
+  public partialQueries: string[];
+  public suggestionRanking: number;
+  public suggestions: string[];
+  public partialQuery: string;
+
+  constructor() {
+    this.partialQueries = [];
+    this.suggestions = [];
+  }
+
+  public buildCustomDataForPartialQueries(index?: number, suggestions?: string[]): IAnalyticsOmniboxSuggestionMeta {
+    this.partialQuery = last(this.partialQueries);
+    this.suggestions = suggestions || this.suggestions;
+    this.suggestionRanking = index || this.suggestionRanking;
+
+    return {
+      partialQueries: this.cleanCustomData(this.partialQueries),
+      suggestionRanking: this.suggestionRanking,
+      suggestions: this.cleanCustomData(this.suggestions),
+      partialQuery: this.partialQuery
+    };
+  }
+
+  public dumpInfo() {
+    console.log(this.partialQueries);
```

Comment:
What is the purpose of that method, is it made to be shown to the client? If yes, use the `Logger` library that is used elsewhere in the code base, this way it can be disabled/enabled if needed.

Reply:
Yeah that was for debugging.
I will delete it